var/spack/repos/builtin/packages/rust/package.py

```diff
@@ -64,6 +71,7 @@ class Rust(Package):
     depends_on('openssl')
     depends_on('libssh2')
     depends_on('libgit2')
+    depends_on('cuda', when='extra_targets=nvptx-nvidia-cuda')
```

Comment:
Typo:
```suggestion
    depends_on('cuda', when='extra_targets=nvptx64-nvidia-cuda')
```

Reply:
Just to check, what does the Rust toolchain need from `cuda`?

Reply:
AFAIK, calling a global function (`extern "ptx-kernel"`) will not be possible without `cuda`. I guess its not a hard dependency, but I would expect the library to be there when I am targeting PTX

Reply:
Hm... I guess I disagree with the notion that a package should create a transitive dependency on another package just for convenience. At most, it should be a `type='run'` dependency, but I don't know if I even want to go that far. A consequence of making it a dependency is that switching the `cuda` version you are using would result in you having to rebuild the `rust` package, even though it doesn't functionally change anything about how the `rust` package is built or runs. If the cuda dependency is instead applied to the package that is building and loading the kernels, then you only have to rebuild that one package when building for different versions of cuda.

@alalazo Could you weigh in?

Reply:
That makes sense to me; sorry, its been a long day...
I think the original reason was that `llvm` depends on `cuda` when `+cuda`, but that seems to be only for reasons of offloading using OpenMP, and is therefore not applicable to `rust`

Reply:
That's alright! Thank you.